### PR TITLE
docs: fix missing bugfix changelog entries

### PR DESCRIPTION
**Pull request by DanStough**
### Description
`bugfix` is an invalid category for our changelog generation tool. Unfortunately, the tool skipped over these entries and cause them to be missing from out changelog. This PR manually updates the changelog for the missing items originally labeled `bugfix`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -162,6 +171,8 @@ BUG FIXES:
 connect: Terminating gateways with a wildcard service entry should no longer pick up connect services as upstreams. [[GH-13958](https://github.com/hashicorp/consul/issues/13958)]
 * proxycfg: Fixed a minor bug that would cause configuring a terminating gateway to watch too many service resolvers and waste resources doing filtering. [[GH-13012](https://github.com/hashicorp/consul/issues/13012)]
 * raft: upgrade to v1.3.8 which fixes a bug where non cluster member can still be able to participate in an election. [[GH-12844](https://github.com/hashicorp/consul/issues/12844)]
+* rpc: Adds a deadline to client RPC calls, so that streams will no longer hang
+indefinitely in unstable network conditions. [[GH-8504](https://github.com/hashicorp/consul/issues/8504)] [[GH-11500](https://github.com/hashicorp/consul/issues/11500)]
```
Comment (eculver) [inline CHANGELOG.md:175]: Is it accurate for this to have both links to #8504 and #11500?

File: CHANGELOG.md [hunk 2]
```diff
@@ -48,6 +52,8 @@ IMPROVEMENTS:
 * envoy: adds additional Envoy outlier ejection parameters to passive health check configurations. [[GH-14238](https://github.com/hashicorp/consul/issues/14238)]
 * metrics: add labels of segment, partition, network area, network (lan or wan) to serf and memberlist metrics [[GH-14161](https://github.com/hashicorp/consul/issues/14161)]
 * snapshot agent: **(Enterprise only)** Add support for path-based addressing when using s3 backend.
+* envoy: validate name before deleting proxy default configurations. [[GH-14290](https://github.com/hashicorp/consul/issues/14290)]
+* rpc: Adds max jitter to client deadlines to prevent i/o deadline errors on blocking queries [[GH-14233](https://github.com/hashicorp/consul/issues/14233)]
```
Comment (eculver) [inline CHANGELOG.md:56]: should these be in the `BUG FIXES:` section?

File: CHANGELOG.md [hunk 3]
```diff
@@ -274,7 +287,9 @@ IMPROVEMENTS:
 * agent: bump default min version for connections to TLS 1.2 [[GH-12522](https://github.com/hashicorp/consul/issues/12522)]
 * api: add QueryBackend to QueryMeta so an api user can determine if a query was served using which backend (streaming or blocking query). [[GH-12791](https://github.com/hashicorp/consul/issues/12791)]
 * ci: include 'enhancement' entry type in IMPROVEMENTS section of changelog. [[GH-12376](https://github.com/hashicorp/consul/issues/12376)]
+* ui: Ensure proxy instance health is taken into account in Service Instance Listings [[GH-12279](https://github.com/hashicorp/consul/issues/12279)]
```
Comment (eculver) [inline CHANGELOG.md:290]: Same here and below. I would expect these all to fall under the `BUG FIXES:` heading.

File: CHANGELOG.md (hunk 1 above)
Comment (eculver) [reply CHANGELOG.md:175]: apologies, you can ignore, I see this in the original entry, so I guess it's valid!

File: CHANGELOG.md (hunk 2 above)
Comment (DanStough) [reply CHANGELOG.md:56]: great catch!

**Review by eculver — APPROVED** [review]
LGTM!

**Pull request merged**